pkg/v1/remote/image.go

```diff
@@ -42,30 +42,63 @@ type remoteImage struct {
 	config       []byte
 }
 
+type ImageOption func(*ImageOpener) error
+
 var _ partial.CompressedImageCore = (*remoteImage)(nil)
 
-// Image accesses a given image reference over the provided transport, with the provided authentication.
-func Image(ref name.Reference, auth authn.Authenticator, t http.RoundTripper) (v1.Image, error) {
-	scopes := []string{ref.Scope(transport.PullScope)}
-	tr, err := transport.New(ref.Context().Registry, auth, t, scopes)
+type ImageOpener struct {
+	auth      authn.Authenticator
+	transport http.RoundTripper
+	ref       name.Reference
+	client    *http.Client
+}
+
+func (i *ImageOpener) Open() (v1.Image, error) {
+	tr, err := transport.New(i.ref.Context().Registry, i.auth, i.transport, transport.DefaultScopes(i.ref))
 	if err != nil {
 		return nil, err
 	}
-	img, err := partial.CompressedToImage(&remoteImage{
-		ref:    ref,
+	ri := &remoteImage{
+		ref:    i.ref,
 		client: &http.Client{Transport: tr},
-	})
+	}
+	imgCore, err := partial.CompressedToImage(ri)
 	if err != nil {
-		return nil, err
+		return imgCore, err
 	}
-	// Wrap the v1.Layers returned by this v1.Image in a hint for downstream
-	// remote.Write calls to facilitate cross-repo "mounting".
 	return &mountableImage{
-		Image:      img,
-		Repository: ref.Context(),
+		Image:      imgCore,
+		Repository: i.ref.Context(),
+	}, nil
+}
+
+func NewImageOpener(ref name.Reference) (*ImageOpener, error) {
+	auth, err := authn.DefaultKeychain.Resolve(ref.Context().Registry)
+	if err != nil {
+		return nil, err
+	}
+	return &ImageOpener{
+		auth:      auth,
```

Comment:
We should default to `authn.Anonymous`.

Reply:
I don't have super strong feelings about this, but does it make more sense to give a default authenticator pointed at the registry the user provides? I'm fine defaulting to anonymous auth instead, but just wondering if it might be a better experience to try and do basic auth for them.